### PR TITLE
cloud-docs/run-environment: Don't overload the term "run"

### PR DESCRIPTION
**Pull request by apparentlymart**
### What

I have revised the "Concurrency and Run Queuing" section of the "Run Environment" page in the HCP Terraform documentation, to avoid overloading the meaning of "run".

[Rendered Preview](https://terraform-docs-common-git-cloud-docs-job-queue-9da94a-hashicorp.vercel.app/terraform/cloud-docs/run/run-environment#concurrency-and-job-queuing)

### Why

This "Run Environment" page belongs to a collection of pages that together describe our capital-R concept of "Run" in HCP Terraform, which refers to the combination of a plan phase and an optional apply phase. "Normal" runs form a per-workspace queue where each run must be resolved (its plan either approved or declined) before proceeding with the next.

(That per-workspace queue of runs is _not_ the same as the global queue of "jobs" that this section is discussing: a run that has already completed its planning and is waiting for approval does not exist in the job queue until someone approves the plan, causing the plan to be applied.)

This particular section was previously using "run" both in its usual meaning _and_ with a new meaning of "job that can be picked up by a worker", which made this section hard to follow when considered in relation to everything else in this chapter of the documentation.

The prior text was already using the word "job" once to describe what was being taken from queues, and so here I've revised the text to consistently use that noun to refer to the kind of thing that is constrained by our concurrency limits. I think that explains the situation a little more clearly while avoiding getting _too_ deep into the implementation details.

I also made the separate editorial decision that jobs take actions and so what they do should be described actively using e.g. the gerund "applying" instead of the nouned-verb "applies" (which is ambiguous with a verb as in "the agent _applies_ the plan", making the prior sentence harder to parse).

Otherwise, I attempted to keep the message largely the same as it was before, but just using hopefully-clearer terminology.

Since the heading text also changed, I tincluded an additional anchor with the previously-used section slug to catch any existing links directly to this section.

----------

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages. (n/a)
- [x] API documentation and the API Changelog have been updated.  (n/a)
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.). (n/a)
- [x] Pages with related content are updated and link to this content when appropriate. (n/a)
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages. (n/a)
- [x] New pages have metadata (page name and description) at the top. (n/a)
- [x] New images are 2048 px wide. (n/a)
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars. (n/a)
- [x] UI elements (button names, page names, etc.) are bolded. (n/a)
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/run/run-environment.mdx [hunk 1]
```diff
@@ -29,17 +29,19 @@ If you are using the SaaS version of HCP Terraform, this means your VCS provider
 
 Terraform Enterprise instances must have network connectivity to any connected VCS providers or managed infrastructure providers.
 
-## Concurrency and Run Queuing
+<a name="concurrency-and-run-queuing"></a>
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/run/run-environment.mdx:32]:
```suggestion
<a name="concurrency-and-run-queuing"></a>
```
We can do this in another PR, but instead of adding this anchor, we should search for the links to this heading update them.

File: website/docs/cloud-docs/run/run-environment.mdx [hunk 2]
```diff
@@ -29,17 +29,19 @@ If you are using the SaaS version of HCP Terraform, this means your VCS provider
...
 
-HCP Terraform uses multiple concurrent worker VMs, which take jobs from a global queue of runs that are ready for processing. (This includes confirmed applies, and plans that have just become the current run on their workspace.)
+## Concurrency and Job Queuing
 
-If the global queue has more runs than the workers can handle at once, some of them must wait until a worker becomes available. When the queue is backed up, HCP Terraform gives different priorities to different kinds of runs:
+HCP Terraform uses multiple concurrent worker VMs, which take jobs from a global queue of operations that are ready to execute. (This includes both creating new plans and then applying those plans once they've been approved.)
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/run/run-environment.mdx:36]:
```suggestion
HCP Terraform uses multiple concurrent worker VMs, which take jobs from a global queue of operations that are ready to execute. This includes both creating new plans and then applying those plans once they've been approved.
```
Per the style guide, avoid using parentheses in prose.

File: website/docs/cloud-docs/run/run-environment.mdx [hunk 3]
```diff
@@ -29,17 +29,19 @@ If you are using the SaaS version of HCP Terraform, this means your VCS provider
...
+If the global queue has more jobs than the workers can handle at once, some of them must wait until a worker becomes available. When the queue is backed up, HCP Terraform gives different priorities to different kinds of jobs:
 
-HCP Terraform can also delay some runs in order to make performance more consistent across organizations. If an organization requests a large number of runs at once, HCP Terraform queues some of them immediately, and delays the rest until some of the initial batch have finished; this allows every organization to continue performing runs even during periods of especially heavy load.
+- Applying approved plans has the highest priority.
+- Creation of normal plans has the next-highest priority.
+- Creation of speculative plans has the lowest priority.
+
+HCP Terraform can also delay some jobs to make performance more consistent across organizations. If an organization requests a large number of jobs at once, HCP Terraform queues some of them immediately, and delays the rest until some of the initial batch have finished; this allows every organization to continue executing jobs even during periods of especially heavy load.
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/run/run-environment.mdx:44]:
```suggestion
HCP Terraform can also delay some jobs to make performance more consistent across organizations. If an organization requests a large number of jobs at once, HCP Terraform queues some of them immediately and delays the rest until some of the initial batch have finished. Doing so allows every organization to continue executing jobs even during periods of especially heavy load.
```
I don't think the semicolon is wrong here, but I'd rather use two sentences to keep it simple.

**Review by trujillo-adam — APPROVED** [review]
Thanks for taking a _run_ at this one!

**Comment by apparentlymart** [comment]
Sorry I missed the feedback on this before! Based on the dates here, I think I was in my last days at HashiCorp when you posted those comments, so I guess I was distracted by other things. I'm glad it was eventually useful. :grinning:
